### PR TITLE
[RISCV] Add scheduling model for Syntacore SCR4 and SCR5 

### PR DESCRIPTION
**Pull request by asi-sc**
Syntacore SCR4 is a microcontroller-class processor core that has much in common
with SCR3, but also supports F and D extensions.
Overview: https://syntacore.com/products/scr4

Syntacore SCR5 is an entry-level Linux-capable 32/64-bit RISC-V processor core
which scheduling model almost match SCR4.
Overview: https://syntacore.com/products/scr5

Co-authored-by: Dmitrii Petrov <dmitrii.petrov@syntacore.com>
Co-authored-by: Anton Afanasyev <anton.afanasyev@syntacore.com>

### CODE REVIEW

**Review by michaelmaitland — APPROVED** [review]
LGTM

**Review by dtcxzyw — APPROVED** [review]
LGTM.

File: llvm/lib/Target/RISCV/RISCVSchedSyntacoreSCR345.td [hunk 1]
```diff
@@ -0,0 +1,449 @@
+//==- RISCVSchedSyntacoreSCR345.td - Syntacore SCR3, SCR4, SCR5 Scheduling Definitions -*- tablegen -*-=//
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVSchedSyntacoreSCR345.td:1]: Line exceeds 80 characters

Comment (asi-sc) [reply llvm/lib/Target/RISCV/RISCVSchedSyntacoreSCR345.td]: Addressed. Also fixed all other long lines in this file.

**Review by topperc — APPROVED** [review]
LGTM

**Review by el-sc — APPROVED** [review]

**Pull request merged**